be/src/storage/lake/tablet_manager.cpp

```diff
@@ -424,11 +417,11 @@ Status TabletManager::put_tablet_metadata(const TabletMetadata& metadata) {
 
 StatusOr<TabletMetadataPtr> TabletManager::load_tablet_metadata(const string& metadata_location, bool fill_cache) {
     auto t0 = butil::gettimeofday_us();
-    MetaFileReader reader(metadata_location, fill_cache);
-    RETURN_IF_ERROR(reader.load());
-    auto res = reader.get_meta();
+    auto metadata = std::make_shared<TabletMetadataPB>();
+    ProtobufFile file(metadata_location);
+    RETURN_IF_ERROR(file.load(metadata.get(), fill_cache));
     g_get_tablet_metadata_latency << (butil::gettimeofday_us() - t0);
-    return res;
+    return std::move(metadata);
```

Comment:
```suggestion
    return metadata;
```

Reply:
any reason?

Reply:
auto return value optimization？

Reply:
> auto return value optimization？

The function return value is of type StatusOr<std::shared_ptr<TabletMetadataPB>>, not std::shared_ptr<TabletMetadataPB>